app/depict/src/main/java/org/openscience/cdk/depict/Depiction.java

```diff
@@ -0,0 +1,322 @@
+        final List<String> formats = new ArrayList<>();
+        formats.add(SVG_FMT);
+        formats.add(SVG_FMT.toUpperCase(Locale.ROOT));
+        formats.add(PS_FMT);
+        formats.add(PS_FMT.toUpperCase(Locale.ROOT));
+        formats.add(PDF_FMT);
+        formats.add(PDF_FMT.toUpperCase(Locale.ROOT));
+        formats.addAll(Arrays.asList(ImageIO.getWriterFormatNames()));
+        return formats;
+    }
+
+    /**
+     * Write the depiction to the provided output stream.
+     *
+     * @param fmt format
+     * @param out output stream
+     * @throws IOException depiction could not be written, low level IO problem
+     * @see #listFormats()
+     */
+    public final void writeTo(String fmt, OutputStream out) throws IOException {
+        if (fmt.equalsIgnoreCase(SVG_FMT)) {
+            out.write(toSvgStr().getBytes(Charsets.UTF_8));
+        } else if (fmt.equalsIgnoreCase(PS_FMT)) {
+            out.write(toEpsStr().getBytes(Charsets.UTF_8));
+        } else if (fmt.equalsIgnoreCase(PDF_FMT)) {
+            out.write(toPdfStr().getBytes(Charsets.UTF_8));
+        } else {
+            ImageIO.write(toImg(), fmt, out);
+        }
+    }
+
+    /**
+     * Write the depiction to the provided output stream.
+     *
+     * @param fmt  format
+     * @param file output destination
+     * @throws IOException depiction could not be written, low level IO problem
+     * @see #listFormats()
+     */
+    public final void writeTo(String fmt, File file) throws IOException {
+        try (FileOutputStream out = new FileOutputStream(file)) {
+            writeTo(fmt, out);
+        }
+    }
+
+    /**
+     * Write the depiction to the provided file path.
+     *
+     * @param fmt  format
+     * @param path output destination path
+     * @throws IOException depiction could not be written, low level IO problem
+     * @see #listFormats()
+     */
+    public final void writeTo(String fmt, String path) throws IOException {
+        writeTo(fmt, new File(replaceTildaWithHomeDir(ensureSuffix(path, fmt))));
+    }
+
+    /**
+     * Write the depiction to the provided file path, the format is determined
+     * by the path suffix.
+     *
+     * @param path output destination path
+     * @throws IOException depiction could not be written, low level IO problem
+     * @see #listFormats()
+     */
+    public final void writeTo(String path) throws IOException {
+        final int i = path.lastIndexOf(DOT);
+        if (i < 0 || i + 1 == path.length())
+            throw new IOException("Can not find suffix in provided path: " + path);
+        final String fmt = path.substring(i + 1);
+        writeTo(fmt, path);
+    }
+
+    /**
+     * Utility for resolving paths on unix systems that contain tilda for
+     * the home directory.
+     *
+     * @param path the file system path
+     * @return normalised path
+     */
+    private static String replaceTildaWithHomeDir(String path) {
+        if (path.startsWith("~/"))
```

Comment:
Common on Linux is also "~${account}/" like "~egonw/" ... not entirely sure how to handle that; dunno how System.getProperty() would then work... perhaps, System.getProperty("user.home") + "../" + path.substring(1) ...

Not important... just wanted to comment something...